include/rocksdb/options.h

```diff
@@ -1946,6 +1946,20 @@ struct ReadOptions {
   // Default: false
   bool allow_unprepared_value = false;
 
+  // EXPERIMENTAL
+  //
+  // When enabled, the iterator will opportunistically check whether its current
+  // superversion number matches most recent CF counterpart during Seek(),
+  // Next() and Prev() calls. If not, it will refresh itself to the latest CF
+  // superversion preserving the very same snapshot it's been assigned to for
+  // consistency. This allows the iterator to adapt to updates such as new
+  // memtables, compactions and flushes.
```

Comment:
SuperVersion is an internal concept, and I'm not opposed to mentioning it as long as we sufficiently describe the user impact. I don't think this last sentence is enough detail for most users to understand. The issue is really long-running iterators holding onto memory and storage resources long after they are obsolete. This setting should fix that problem as long as iterators periodically make some progress. The feature is engineered so that the performance impact should be negligible, and we expect the default to be `true` some time in the future.